rust/flatbuffers/src/verifier.rs

```diff
@@ -0,0 +1,615 @@
+                    )?;
+                }
+            }
+        }
+        Ok(())
+    }
+}
+
+impl std::fmt::Display for InvalidFlatbuffer {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "InvalidFlatbuffer: ")?;
+        use InvalidFlatbuffer::*;
+        match self {
+            TooManyTables => write!(f, "Too many tables.")?,
+            ApparentSizeTooLarge => write!(f, "Apparent size too large.")?,
+            DepthLimitReached => write!(f, "Nested table depth limit reached.")?,
+            MissingRequiredField {
+                required,
+                error_trace,
+            } => {
+                write!(f, "Missing required field `{}`.\n{}", required, error_trace)?;
+            }
+            InconsistentUnion {
+                field,
+                field_type,
+                error_trace,
+            } => {
+                write!(
+                    f,
+                    "Union exactly one of union discriminant (`{}`) and value\
+                     (`{}`) are present. .\n{}",
+                    field_type, field, error_trace
+                )?;
+            }
+            BadString {
+                utf8_error,
+                has_null_terminator,
+                range,
+                error_trace,
+            } => {
+                write!(f, "string at range [{}, {}) ", range.start, range.end)?;
+                if !has_null_terminator {
+                    write!(f, "is missing its null terminator")?;
+                    if utf8_error.is_some() {
+                        write!(f, "and ")?;
+                    }
+                }
+                if let Some(e) = utf8_error {
+                    write!(f, "is invalid utf8 ({})", e)?;
+                }
+                write!(f, ".\n{}", error_trace)?;
+            }
+            Unaligned {
+                position,
+                unaligned_type,
+                error_trace,
+            } => {
+                write!(
+                    f,
+                    "Type `{}` at position {:?} is unaligned.\n{}",
+                    unaligned_type, position, error_trace
+                )?;
+            }
+            RangeOutOfBounds { range, error_trace } => {
+                write!(
+                    f,
+                    "Range [{}, {}) is out of bounds.\n{}",
+                    range.start, range.end, error_trace
+                )?;
+            }
+            SignedOffsetOutOfBounds {
+                soffset,
+                position,
+                error_trace,
+            } => {
+                write!(
+                    f,
+                    "Signed Offset at position {:?} has value {:?} \
+                     which points out of bounds.\n{}",
+                    position, soffset, error_trace
+                )?;
+            }
+        }
+        Ok(())
+    }
+}
+
+pub type Result<T> = core::prelude::v1::Result<T, InvalidFlatbuffer>;
```

Comment:
core?

Reply:
I could also use `std::prelude::v1::Result` (or not just not shadow the default result) but this works fine.